helix-core/src/syntax.rs

```diff
@@ -1827,6 +1833,25 @@ fn injection_for_match<'a>(
             language_name = Some(name);
         } else if index == content_capture_index {
             content_node = Some(capture.node);
+        } else if index == language_filename_index {
+            let mut languages = config::default_syntax_loader().language;
+            if let Ok(user_config) = config::user_syntax_loader() {
+                let mut user_languages = user_config.language;
+                languages.append(&mut user_languages);
+            }
+            let file = byte_range_to_str(capture.node.byte_range(), source);
+            let extension = Path::new(file.as_ref()).extension();
+            if let Some(extension) = extension {
+                let language = languages.iter().find(|language| {
+                    language
+                        .file_types
+                        .iter()
+                        .any(|e| extension.to_str() == Some(e))
+                });
+                if let Some(language) = language {
+                    language_name = Some(Cow::from(language.language_id.clone()));
+                }
+            }
```

Comment:
I think this differs slightly from the file-type detection we use for determining the language to use. Can we re-use that code here without too much headache?

In particular there's a case where `Makefile` or `Dockerfile` won't be highlighted because they don't have an extension while they are correctly picked up by the `file-types` configuration ([this](https://github.com/helix-editor/helix/blob/8fc493137d159c4b1fbf964ed52dc6d53f6fcaf8/languages.toml#L784) and [this](https://github.com/helix-editor/helix/blob/8fc493137d159c4b1fbf964ed52dc6d53f6fcaf8/languages.toml#L960)).

Reply:
Hey thanks, the code is a lot simpler to follow now as well, and I didn't realize previously that the syntax struct already had a reference to the loader :facepalm:  so I fixed that as well